### PR TITLE
Add README

### PR DESCRIPTION
**Pull request by call**
This PR encapsulates the following changes:
- Remove `cat`ting of the log file from `macos_puppet_apply.yml` workflow—this was for debugging and is no longer needed in `master`
- Add (reasonably) comprehensive README, using [`puppetlabs/ntp`](https://forge.puppet.com/puppetlabs/ntp) and [`puppetlabs/apache`](https://forge.puppet.com/puppetlabs/apache) as best-practice examples
- Change the name of the GitHub Actions workflow from _MacOS puppet apply_ to _MacOS QA_

### CODE REVIEW

**Comment by call** [comment]
cc @philipdelorenzo @dquills

File: README.md [hunk 1]
```diff
@@ -1,87 +1,105 @@
...
-    * [Setup requirements](#setup-requirements)
-    * [Beginning with buildkite_agent](#beginning-with-buildkite_agent)
-3. [Usage - Configuration options and additional functionality](#usage)
-4. [Limitations - OS compatibility, etc.](#limitations)
-5. [Development - Guide for contributing to the module](#development)
+- [buildkite_agent](#buildkiteagent)
+  - [Table of Contents](#table-of-contents)
+  - [Description](#description)
+  - [Setup](#setup)
+    - [Beginning with buildkite_agent](#beginning-with-buildkiteagent)
```
Comment (philipdelorenzo) [inline README.md]:
```suggestion
    - [Beginning with buildkite_agent](#beginning-with-buildkite_agent)
```

File: README.md [hunk 2]
```diff
@@ -1,87 +1,105 @@
...
 
-1. [Description](#description)
-2. [Setup - The basics of getting started with buildkite_agent](#setup)
-    * [What buildkite_agent affects](#what-buildkite_agent-affects)
-    * [Setup requirements](#setup-requirements)
-    * [Beginning with buildkite_agent](#beginning-with-buildkite_agent)
-3. [Usage - Configuration options and additional functionality](#usage)
-4. [Limitations - OS compatibility, etc.](#limitations)
-5. [Development - Guide for contributing to the module](#development)
+- [buildkite_agent](#buildkiteagent)
```
Comment (philipdelorenzo) [inline README.md]:
```suggestion
- [buildkite_agent](#buildkite_agent)
```

**Review by philipdelorenzo — CHANGES_REQUESTED** [review]
I made two tiny suggestions for some of the anchors that didn't resolve.

File: README.md (hunk 1 above)
Comment (call) [reply README.md]: Good eyes! 👀 As it turns out though, underscores are [not recognized in markdown heading anchors](https://gist.github.com/asabaylus/3071099#gistcomment-1593627). News to me! 🗞

Comment (philipdelorenzo) [reply README.md]: Dang, it works in the master (that's why I placed them in the suggestion).

Comment (call) [reply README.md]: Of note, though `pdk new module` generates markdown anchors with underscores (e.g. in the Table of Contents), VSCode automatically removes the underscores upon save.

File: README.md (hunk 2 above)
Comment (call) [reply README.md]: See above comment; underscores don't belong in markdown heading anchor tags.

File: README.md (hunk 1 above)
Comment (philipdelorenzo) [reply README.md]: Oh hahaha, that makes sense.

**Comment by call** [comment]
@philipdelorenzo 

Ready for re-review. Thanks!

File: README.md (hunk 1 above)
Comment (call) [reply README.md]: @philipdelorenzo prodded me further on this, and it turns out that there was some overly-aggressive VSCode auto-formatting happening.

TL;DR - Underscores in heading anchors == 👌

**Review by philipdelorenzo — APPROVED** [review]
Anchors are working now!
:shipit:
